fulp_modules/devils/devil_powers.dm

```diff
@@ -0,0 +1,160 @@
+	action_icon_state = "pitchfork"
+	action_background_icon_state = "bg_demon"
+
+/obj/item/pitchfork/demonic
+	name = "demonic pitchfork"
+	desc = "A red pitchfork, it looks like the work of the devil."
+	lefthand_file = 'fulp_modules/devils/devil_icon/devils_lefthand.dmi'
+	righthand_file = 'fulp_modules/devils/devil_icon/devils_righthand.dmi'
+	force = 19
+	throwforce = 24
+	light_system = MOVABLE_LIGHT
+	light_range = 3
+	light_power = 6
+	light_color = COLOR_SOFT_RED
+
+/obj/item/pitchfork/demonic/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/two_handed, force_unwielded=19, force_wielded=25)
+
+/obj/item/pitchfork/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] impales [user.p_them()]self in [user.p_their()] abdomen with [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	return (BRUTELOSS)
+
+/obj/item/pitchfork/demonic/ascended/afterattack(atom/target, mob/user, proximity)
+	. = ..()
+	if(!proximity || !wielded)
+		return
+	if(iswallturf(target))
+		var/turf/closed/wall/W = target
+		user.visible_message("<span class='danger'>[user] blasts \the [target] with \the [src]!</span>")
+		playsound(target, 'sound/magic/disintegrate.ogg', 100, TRUE)
+		W.break_wall()
+		W.ScrapeAway(flags = CHANGETURF_INHERIT_AIR)
+		return
+
+
+/// Summon Violin
+/obj/effect/proc_holder/spell/targeted/conjure_item/violin
+	item_type = /obj/item/instrument/violin/golden/devil
+	desc = "A devil's instrument of choice.  Use this to summon/unsummon your golden violin."
+	invocation_type = INVOCATION_WHISPER
+	invocation = "I ain't had this much fun since Georgia."
+	action_icon_state = "golden_violin"
+	name = "Summon golden violin"
+	action_icon = 'fulp_modules/devils/devil_icon/devil_icons.dmi'
+	action_background_icon_state = "bg_demon"
+
+/obj/item/instrument/violin/golden/devil
+	name = "golden violin"
+	desc = "A golden musical instrument with four strings and a bow. \"The devil went down to space, he was looking for an assistant to grief.\""
+	icon = 'fulp_modules/devils/devil_icon/violin.dmi'
+	lefthand_file = 'fulp_modules/devils/devil_icon/devils_lefthand.dmi'
+	righthand_file = 'fulp_modules/devils/devil_icon/devils_righthand.dmi'
+	icon_state = "golden_violin"
+	inhand_icon_state = "golden_violin"
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+
+/// Summon Fireball
+/obj/effect/proc_holder/spell/aimed/fireball/hellish
+	name = "Hellfire"
+	desc = "This spell launches hellfire at the target."
+
+	school = "evocation"
+	charge_max = 80
+	clothes_req = FALSE
+	invocation = "Your very soul will catch fire!"
+	invocation_type = INVOCATION_SHOUT
+	range = 2
+
+	projectile_type = /obj/projectile/magic/aoe/fireball/infernal
+
+	action_background_icon_state = "bg_demon"
+
+/obj/projectile/magic/aoe/fireball/infernal
+	name = "infernal fireball"
+	exp_heavy = -1
+	exp_light = -1
+	exp_flash = 4
+	exp_fire= 5
+
+/obj/projectile/magic/aoe/fireball/infernal/on_hit(target)
+	. = ..()
```

Comment:
remember that you're calling the fireball parent code here

so check to see if the antimagic check is redundant, and if i should start at 10 instead of 0